PowerEditor/src/WinControls/ColourPicker/WordStyleDlg.cpp

```diff
@@ -817,6 +817,34 @@ bool WordStyleDlg::selectThemeByName(const TCHAR* themeName)
 	return true;
 }
 
+bool WordStyleDlg::goToSection(const TCHAR* sectionNames)
+{
+	if (!sectionNames || !sectionNames[0])
+		return false;
+
+	std::vector<generic_string> sections = tokenizeString(sectionNames, ':');
+
+	if (sections.size() == 0 || sections.size() >= 3)
+		return false;
+
+	auto i = ::SendDlgItemMessage(_hSelf, IDC_LANGUAGES_LIST, LB_FINDSTRING, (WPARAM)-1, (LPARAM)sections[0].c_str());
+	if (i == LB_ERR)
+		return false;
+	::SendDlgItemMessage(_hSelf, IDC_LANGUAGES_LIST, LB_SETCURSEL, 0, i);
+	setStyleListFromLexer(static_cast<int>(i));
+
+	if (sections.size() == 1)
+		return true;
+
+	i = ::SendDlgItemMessage(_hSelf, IDC_STYLES_LIST, LB_FINDSTRING, (WPARAM)-1, (LPARAM)sections[1].c_str());
+	if (i == LB_ERR)
+		return false;
+	::SendDlgItemMessage(_hSelf, IDC_STYLES_LIST, LB_SETCURSEL, 0, i);

```

Comment:
@ozone10 
This line doesn't take effect. I would appreciate if you can take a look on it.

Reply:
Problem found:
It should be "(_hSelf, IDC_STYLES_LIST, LB_SETCURSEL, i, 0);" instead of "(_hSelf, IDC_STYLES_LIST, LB_SETCURSEL, 0, i);"
I just copied the existent code - it means there are some bugs....

Reply:
@donho 
I was away. IIRC WPARAM is used for handle and number value, and LPARAM for pointer. This should help next time.